noahfruttarol/allergy_aware
# Release 1.1.0: Final Presentation

* Add title for supabase data

* Update README.md

* Backend project structure

* Project structure and Authentication

* Fix import

* Add dependencies directory with get_user.py for user authentication

* Add schemas directory with Pydantic models for dishes, ingredients and restaurants

* Add config.py for connecting to Supabase

* Add main.py with API setup using FastAPI

* Add router.py to set up API routes for allergens and authentication

* Add auth.py with login endpoint for user authentication

* fix authentication

* Add search endpoint to search for Ingredients and allergens

* Documentation

* Edited README file

* Added required packages to requirements.txt

* Fix requirements.txt

* Fixed env load

* Fix Supabase requests

* Fix Supabase request

* Add load env

* Fix Exceptions

* Fix supabase logic

* Fixed search endpoint

* Move frontend to different directory, add separate readme

* Fixed search

* Ignore all environments

* Initial frontend search setup

* Docker setup and config.js for backend URL

* Setup basic login

* Edit comments

* Added CORS middleware

* Fixing API endpoints

* Remove supabase client initialization

* SearchPage.jsx logic

* Update package.json and package-lock.json files

* Fix styling

* After successful login redirect user

* Fix token access

* Replace usage of config

* Cleanup README.md and gitignore

* Fixed first query

* logging

* test hardcode

* Implement fix

* further tests

* Potential fix

* Remove logs

* Test

* Additional tests

* Add cors handling for backend

* test adding --proxy-headers

* Try template nginx custom config

* wrap in http

* undo

* Check frontend url loaded correctly

* Remove space

* Cleanup frontend for better usability

* Nicer loading around the place

* Setup traefik reverse proxy + load balancer

* Adjust ports

* Adjust backend

* Revert

* Fix dockerfile environment for frontend

* Absolute address

* Try https

* Back to http

* Accept from all IPv6 addresses

* Use public api endpoint

* Additional logs

* More log

* Add frontend url to dockerfile

* Setup logging

* Adjust host

* Fix request issue

* Adjust listening ports and OAuth2 password bearer token url adjustment

* Better handling of redirect

* Prevent page change on login

* Update repo readme

* Update README.md

* Update index.html title of page

* Create LICENSE.md

* modify search functionality (#16)

* Change search to searching by dish name

* edit frontend search page

* updated frontend (#15)

* updated frontend

* Updated search page, updated landing page

* Adjust footer, index.html, favicon and cleanup

---------

Co-authored-by: Josh Machado <joshomac505@gmail.com>

* Add all dishes and searching by categories

* Edit Page (#17)

* updated frontend

* Cleanup and edit page setup

* Setup edit page UI to play around with

* Setup edit page and backend

* Couple bugfixes

* Updated search page, updated landing page

* Adjust footer, index.html, favicon and cleanup

* Cleanup UI and make consistent

* Add edit page link

* landing page updates

* landing css changes

* Landing mobile optimizations

* mobile optimizations for search page

* Landing page image changes

* Fix merge lol

* Allow for creation of ingredients, allergens and deletion of dishes

* Cleanup UI

* comment unnecessary

* Selecting an existing ingredient should pull in the allergens
    Ensure no duplicate entries (sanitize function)

---------

Co-authored-by: Nathan7796 <nathan@cabell.ca>

* fix import

* Allow creation/editing of categories (#18)

* updated frontend

* Cleanup and edit page setup

* Setup edit page UI to play around with

* Setup edit page and backend

* Couple bugfixes

* Updated search page, updated landing page

* Adjust footer, index.html, favicon and cleanup

* Cleanup UI and make consistent

* Add edit page link

* landing page updates

* landing css changes

* Landing mobile optimizations

* mobile optimizations for search page

* Landing page image changes

* Fix merge lol

* Allow for creation of ingredients, allergens and deletion of dishes

* Cleanup UI

* comment unnecessary

* Selecting an existing ingredient should pull in the allergens
    Ensure no duplicate entries (sanitize function)

* Insertion of categories in edit page

* Stability changes from bugs

* Adjust color of allergen tag & size of results page

---------

Co-authored-by: Nathan7796 <nathan@cabell.ca>

* Add logo & enhance navbar (#19)

* Add new logos, enhance navbar + mobile

* Move image location

* Move image online

---------

Co-authored-by: Salma Yousef <salmayousef@uvic.ca>
Co-authored-by: Salmayousef <103972196+Salmayousef@users.noreply.github.com>
Co-authored-by: Nathan7796 <127626675+Nathan7796@users.noreply.github.com>
Co-authored-by: Nathan7796 <nathan@cabell.ca>
